### ISSUE TITLE
ETag can't work for double type

### ISSUE THREAD

**Issue by xuzhg** [OData/WebApi#475]
See the related information: https://github.com/OData/ODataSamples/issues/30

**Comment by congysu** [comment]
Can we use the ConvertFromUriLiteral with reference to model:
```c#
        public static object ConvertFromUriLiteral(string value, ODataVersion version, IEdmModel model, IEdmTypeReference typeReference)
```
https://github.com/OData/odata.net/blob/master/src/Microsoft.OData.Core/Query/ODataUriUtils.cs#L49


```c#
                object obj = ODataUriUtils.ConvertFromUriLiteral(valueString, ODataVersion.V4);
```
https://github.com/OData/WebApi/blob/f4573605bd88dabe882c4dc45e4a4ed0e55b32e0/OData/src/System.Web.OData/OData/Formatter/DefaultODataETagHandler.cs#L83

**Comment by VikingsFan** [comment]
Merged e12bae709fc49a3ce7a3e458219f82800ea07774
